File: w09/d03/quin_cogdell/mail_objects_app/app/assets/javascripts/package.js
function Package() {
  this.content =[];
  this.fragile = false;
  this.broken = false;
  this.weight = 100;
}


Package.prototype.addContent = function(item) {
  this.content.push(item);
};

Package.prototype.setFragile = function(bool) {
    if (typeof bool === 'boolean'){
      this.fragile = bool;
   } else {
      throw "must be a boolean";
   }
};

Package.prototype.setBroken = function(bool) {
    if (typeof bool === 'boolean'){
      this.isBroken = bool;
   } else {
      throw "must be a boolean";
   }
};

Package.prototype.isBroken = function() {
  return this.broken;
};

Package.prototype.shake = function() {
  if(this.fragile === true) {
    this.broken = true;
  }
};

Package.prototype.isHeavy = function() {
  return this.weight > 10;
};

File: w09/d01/jimmy_davis/mail_items/lib/package.rb
class Package < MailItem

  attr_accessor :is_fragile, :is_broken

  def initialize
    @content = [1,2]
    @is_fragile = false
    @is_broken = false
  end

  def content
    @content
  end

  def shake
    if @is_fragile
      @is_broken = true
    else
      @is_broken = false
    end
  end

  # def is_fragile?
  #   true
  # end

  # def is_broken?
  #   false
  # end

end
